pkg/controller/controller.go

```diff
@@ -127,30 +128,47 @@ func (haproxy *HAProxyController) BackendDefaults() defaults.Backend {
 
 // OnUpdate regenerate the configuration file of the backend
 func (haproxy *HAProxyController) OnUpdate(cfg ingress.Configuration) ([]byte, error) {
-	data, err := haproxy.template.execute(newControllerConfig(&cfg, haproxy))
+	updatedConfig := newControllerConfig(&cfg, haproxy)
+
+	reconfigureBackends(haproxy, updatedConfig)
+
+	data, err := haproxy.template.execute(updatedConfig)
```

Comment:
Just to let you know Ingress now implement it's own equals before call `OnUpdate` and `Reload`  doesn't exist anymore. I can update Ingress right now (you'd need to merge or rebase) or I can update Ingress on top of your code and remove the `configChanged` part afterwards. I'd follow the latter but this definitely is your choice: update ingress now or on top of your changes?